src/script/page/LeftSidebar/panels/StartUI/PeopleTab.tsx

```diff
@@ -0,0 +1,360 @@
+      const searchFields = isHandle ? [SEARCHABLE_FIELDS.USERNAME] : undefined;
+
+      // If the user typed a domain, we will just ignore it when searchng for the user locally
+      const [domainFreeQuery] = query.split('@');
+      const contactResults = searchRepository.searchUserInSet(domainFreeQuery, localSearchSources, searchFields);
+      const connectedUsers = conversationState.connectedUsers();
+      const filteredResults = contactResults.filter(
+        user => connectedUsers.includes(user) || teamRepository.isSelfConnectedTo(user.id) || user.username() === query,
+      );
+
+      const localSearchResults: SearchResultsData = {
+        contacts: filteredResults,
+        groups: conversationRepository.getGroupsByName(query, isHandle),
+        others: [],
+      };
+      setResults(localSearchResults);
+      onSearchResults(localSearchResults);
+      if (canSearchUnconnectedUsers) {
+        try {
+          const userEntities = await searchRepository.searchByName(query, isHandle);
+          const localUserIds = localSearchResults.contacts.map(({id}) => id);
+          const onlyRemoteUsers = userEntities.filter(user => !localUserIds.includes(user.id));
+          const results = userState.self().inTeam()
+            ? await organizeTeamSearchResults(onlyRemoteUsers, localSearchResults, query)
+            : {...localSearchResults, others: onlyRemoteUsers};
+
+          if (currentSearchQuery.current === searchQuery) {
+            // Only update the results if the query that has been processed correspond to the current search query
+            onSearchResults(results);
+            setResults(results);
+          }
+        } catch (error) {
+          if (isBackendError(error)) {
+            if (error.code === HTTP_STATUS.UNPROCESSABLE_ENTITY) {
+              return setHasFederationError(true);
+            }
+            if (error.code === HTTP_STATUS.BAD_REQUEST && error.label === BackendErrorLabel.FEDERATION_NOT_ALLOWED) {
+              return logger.warn(`Error searching for contacts: ${error.message}`);
+            }
+          }
+          logger.error(`Error searching for contacts: ${(error as any).message}`, error);
+        }
+      }
+    },
+    300,
+    [searchQuery],
+  );
+
+  useEffect(() => {
+    // keep track of the most up to date value of the search query (in order to cancel outdated queries)
+    currentSearchQuery.current = searchQuery;
+    return () => {
+      currentSearchQuery.current = '';
+      onSearchResults(undefined);
+    };
+  }, [searchQuery]);
+
+  return (
+    <>
+      {hasFederationError && (
+        <div className="start-ui-fed-domain-unavailable">
+          <div className="start-ui-fed-domain-unavailable__head">{t('searchConnectWithOtherDomain')}</div>
+          <span className="start-ui-fed-domain-unavailable__text">{t('searchFederatedDomainNotAvailable')}</span>
+          {/*FIXME: re-enable when federation article is available
```

Comment:
```suggestion
          {/*@todo: re-enable when federation article is available
```

<img width="439" alt="image" src="https://user-images.githubusercontent.com/63250054/162726840-7a6a19a0-6b78-4143-b903-9c19be9b37c9.png">